pkg/microservice/aslan/core/templatestore/service/chart.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2021 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/27149chen/afero"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	configbase "github.com/koderover/zadig/pkg/config"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/common/service/fs"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/templatestore/repository/models"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/templatestore/repository/mongodb"
+	fsutil "github.com/koderover/zadig/pkg/util/fs"
+)
+
+func GetChartTemplate(name string, logger *zap.SugaredLogger) (*Chart, error) {
+	chart, err := mongodb.NewChartColl().Get(name)
+	if err != nil {
+		logger.Errorf("Failed to get chart template %s, err: %s", name, err)
+		return nil, err
+	}
+
+	localBase := configbase.LocalChartTemplatePath(name)
+	s3Base := configbase.ObjectStorageChartTemplatePath(name)
+	if err = fs.PreloadFiles(name, localBase, s3Base, logger); err != nil {
+		return nil, err
+	}
+
+	base := filepath.Base(chart.Path)
+	localPath := filepath.Join(localBase, base)
+	fis, err := fs.GetFileInfos(os.DirFS(localPath))
+	if err != nil {
+		logger.Errorf("Failed to local chart template %s from path %s.", name, localPath)
```

Comment:
1. Delete the ending dot
2. Failed to get local chart?
3. No error message output